### PR TITLE
Second Round of Jakarta EE 10 Work

### PR DESCRIPTION
**Pull request by ropalka**
This change introduces:

 * Adds Servlet preview implementation

https://issues.redhat.com/browse/UNDERTOW-2052
https://issues.redhat.com/browse/UNDERTOW-2084
https://issues.redhat.com/browse/UNDERTOW-2085
https://issues.redhat.com/browse/UNDERTOW-2086
https://issues.redhat.com/browse/UNDERTOW-2087
https://issues.redhat.com/browse/UNDERTOW-2088
https://issues.redhat.com/browse/UNDERTOW-2089
https://issues.redhat.com/browse/UNDERTOW-2051

### CODE REVIEW

File: core/src/main/java/io/undertow/server/HttpServerExchange.java [hunk 1]
```diff
@@ -336,6 +336,9 @@ public final class HttpServerExchange extends AbstractAttachable {
      */
     private InetSocketAddress destinationAddress;
 
+    private static final AtomicLong REQUEST_ID_GENERATOR = new AtomicLong(0);
```
Comment (boris-unckel) [inline core/src/main/java/io/undertow/server/HttpServerExchange.java:339]: Is it OK from a security perspective to have predictable ids here? (It's only the pattern that is suspicious, not a fact I know for the case).

File: core/src/main/java/io/undertow/server/ServerConnection.java [hunk 2]
```diff
@@ -46,6 +47,17 @@
  */
 public abstract class ServerConnection extends AbstractAttachable implements ConnectedChannel  {
 
+    private static final AtomicLong CONNECTION_ID_GENERATOR = new AtomicLong(0);
```
Comment (boris-unckel) [inline core/src/main/java/io/undertow/server/ServerConnection.java:50]: Is it OK from a security perspective to have predictable ids here? (It's only the pattern that is suspicious, not a fact I know for the case).

File: core/src/main/java/io/undertow/server/HttpServerExchange.java (hunk 1 above)
Comment (fl4via) [reply core/src/main/java/io/undertow/server/HttpServerExchange.java:339]: That's a very good question! From the top of my head, now, it seems it is okay, but I'll do some research before considering this the final implementation for connection ids.

File: core/src/main/java/io/undertow/server/ServerConnection.java (hunk 2 above)
Comment (fl4via) [reply core/src/main/java/io/undertow/server/ServerConnection.java:50]: That's a very good question! From the top of my head, now, it seems it is okay, but I'll do some research before considering this the final implementation for connection ids.

**Pull request merged**